lib/que/locks/active_job_extensions.rb

```diff
@@ -23,13 +23,27 @@ def do_enqueue(job, **job_options)
       # https://github.com/rails/rails/pull/19498)
       if job.respond_to?(:queue_name) && ::ActiveJob.version.segments.first > 4
         job_options[:queue] = job.queue_name
+      else
+        job_options[:queue] = "default"
```

Comment:
Que 2 presumes that job.queue_name will always be available as I it depends on a later version of active job. To keep the behaviour consistent with the older version of active job I needed to provide a queue name when not specified. "default" was consistent to what the old test specified but I am not sure where if anywhere I could get a more appropriate value. This could also be a configurable option but I wasn't sure how to implement this.